pkg/miniogw/integration_test.go

```diff
@@ -142,7 +144,7 @@ func TestUploadDownload(t *testing.T) {
 }
 
 // runGateway creates and starts a gateway
-func runGateway(ctx context.Context, gwCfg config, uplinkCfg uplink.Config, log *zap.Logger, ident *identity.FullIdentity) (err error) {
+func runGateway(ctx context.Context, gwCfg config, uplinkCfg uplink.Config, encKey *storj.Key, log *zap.Logger, ident *identity.FullIdentity) (err error) {
```

Comment:
Is there a reason the encryption key isn't part of the config? To me it seems it should be part of the config rather than a separate argument.

Reply:
Although it might be that we are conflating two different things:

1. the command line arguments to uplink
2. the configuration of the uplink

Reply:
> Is there a reason the encryption key isn't part of the config? To me it seems it should be part of the config rather than a separate argument.

It was my intention but I couldn't make it work due to the inflexibility of `cfgstruct`; I tried to explain that in the `TODO` comments which should be removed before merging it.

> Although it might be that we are conflating two different things:
> 1. the command line arguments to uplink
> 2. the configuration of the uplink

It could be; being honest after spending quite a few time trying to figure it out if I could keep the _key_ in the configuration, I wasn't able to do so if we wanted to withdraw it from the command line arguments, or at least I didn't find the way, so if you know it, it will be welcome.